src/callBuilder.js

```diff
@@ -0,0 +1,139 @@
+import {NotFoundError, NetworkError, BadRequestError} from "./errors";
+
+let URI = require("URIjs");
+let URITemplate = require('URIjs/src/URITemplate');
+
+
+let axios = require("axios");
+var EventSource = (typeof window === 'undefined') ? require('eventsource') : window.EventSource;
+let toBluebird = require("bluebird").resolve;
+
+/**
+* @class Builder
+*/
+export class CallBuilder {
+
+    /*
+    * @constructor
+    */
+    constructor(url) {
+        this.url = url;
+        this.filter = [];
+    }
+
+    checkFilter() {
+        if (this.filter.length >= 2) {
+            throw new BadRequestError("Too many filters specified", this.filter);
+        } 
+        if (this.filter.length === 1) {
+            this.url.segment(this.filter[0]);
+        }        
+    }
+
+    call() {
```

Comment:
This method should get some documentation.  Even a simple one liner comment would be great: "Triggers an http request using this builders current configuration"